### PR TITLE
feat: step towards feature folders for replay backend

### PR DESCRIPTION
**Pull request by pauldambra**
Backend files for replay are spread allover the place. We use feature folders in the front-end code. Let's step towards that for the backend by gathering the replay files in one place in posthog and ee

### CODE REVIEW

File: posthog/session_recordings/realtime_snapshots.py [hunk 1]
```diff
@@ -72,8 +72,8 @@ def get_realtime_snapshots(team_id: str, session_id: str, attempt_count=0) -> Op
     except Exception as e:
         # very broad capture to see if there are any unexpected errors
         capture_exception(
-            "get_realtime_snapshots_failed",
-            extras={"attempt_count": attempt_count},
+            e,
+            extras={"attempt_count": attempt_count, "operation": "get_realtime_snapshots"},
```
Comment (pauldambra) [inline posthog/session_recordings/realtime_snapshots.py:76]: obey mypy

File: posthog/session_recordings/session_recording_helpers.py
Comment (pauldambra) [inline posthog/session_recordings/session_recording_helpers.py:1]: mypy went to town in this file...

We should probably switch to operate entirely on lists or entirely on generators when we delete the old session recording code

File: posthog/session_recordings/session_recording_helpers.py [hunk 2]
```diff
@@ -149,16 +151,22 @@ def preprocess_replay_events(events: List[Event], max_size_bytes=1024 * 1024) ->
     3. If not, we split out the "full snapshots" from the rest (they are typically bigger) and send them individually, trying one more time to group the rest, otherwise sending them individually
     """
 
+    if isinstance(_events, Generator):
+        # we check the first item in the events below so need to be dealing with a list
+        events = list(_events)
+    else:
+        events = _events
```
Comment (pauldambra) [inline posthog/session_recordings/session_recording_helpers.py:158]: we sometimes pass a generator into this function, and then check `events[0]` but generators aren't subscriptable... so mypy was very unhappy with us

**Review by benjackwhite — APPROVED** [review]
The git file length diff is odd  (+4133, -195). Wondering if there is a file that should have been removed that isn't or a snapshot or something? 

Other than this is great tidying 👍

**Comment by pauldambra** [comment]
@fuziontech this doesn't change anything about the session recording/replay migrations, just moves them. I've checked I can nuke and recreate locally

**Pull request merged**